pkg/client/images.go

```diff
@@ -61,3 +62,10 @@ func (c *images) Delete(name string) (err error) {
 	err = c.r.Delete().Resource("images").Name(name).Do().Error()
 	return
 }
+
+// UpdateSignatures allows to modify signatures of given image.
+func (c *images) UpdateSignatures(image *imageapi.Image) (result *imageapi.Image, err error) {
```

Comment:
(I have only a vague idea about what named subresources and this discussion is about…)

Deleting signatures is not quite essential, but could be very useful.  (We do need at least one of deleting signatures and untagging images, to handle the “image is known to have a vulnerability and is no longer allowed” case.)

If we need a unique identifier, moving this responsibility to clients (who don’t have any way to coordinate setting non-colliding names that is not available to the server; if anything, they have fewer ways to do so) seems to put an undue burden on them.

Wouldn’t it be easier for the API server to assign a UUID on the first upload of a signature? Clients would upload signatures without specifying an UUID, clients who upload a duplicate signature would be silently ignored. The UUIDs would be available when listing signatures (a single API call to get all signatures and their UUIDs, the common case is to need to know all of the signatures), and _only_ used for deleting signatures: Copying an image from one server to another would not preserve these UUIDs.